backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java

```diff
@@ -208,12 +208,15 @@ private void validateEnvironmentVariableName(String name) throws ModelDBExceptio
   private String computeSHA(EnvironmentBlob blob) throws NoSuchAlgorithmException {
     StringBuilder sb = new StringBuilder();
     sb.append("env:");
-    for (EnvironmentVariablesBlob environmentVariablesBlob : blob.getEnvironmentVariablesList()) {
-      sb.append(":name:")
-          .append(environmentVariablesBlob.getName())
-          .append("value:")
-          .append(environmentVariablesBlob.getValue());
-    }
+    List<
+            ai.verta.modeldb.versioning.autogenerated._public.modeldb.versioning.model
+                .EnvironmentVariablesBlob>
+        environmentVariables =
+            ai.verta.modeldb.versioning.autogenerated._public.modeldb.versioning.model
+                .EnvironmentBlob.fromProto(blob)
+                .EnvironmentVariables;
+    sb.append(
+        environmentVariables == null ? "null" : new HashSet<>(environmentVariables).hashCode());
```

Comment:
why you are set to null as a string in StringBuilder? here something wrong. you should just leave it if it is null.

Reply:
fixed